### ISSUE TITLE
Credo.Check.Refactor.CondStatements false positive?

### ISSUE THREAD

**Issue by fuelen** [rrrene/credo#816]
### Environment

* Credo version (`mix credo -v`): 1.4.0
* Erlang/Elixir version (`elixir -v`): 1.10.4/ OTP 22
* Operating system: Linux

### What were you trying to do?
```elixir
cond do
  admin?(user) -> :apply1
  moderator?(user) -> :apply2
end
```

### Expected outcome
successful pass, because recommended way of writing this as
```elixir
if admin?(user) do
  :apply1
else
  :apply2
end
```
is not the same what I'm trying to achieve.
In the first code snippet Elixir will raise CondClauseError if user has another role.
This is an expected result. I don't want run `:apply2` for another type of user.

I suggest triggering `Credo.Check.Refactor.CondStatements` only if the last statement is `true`.

### Actual outcome
`Credo.Check.Refactor.CondStatements` is triggered

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in `1.5.0-rc.5`.

**Comment by fuelen** [comment]
thank you!

**Comment by fuelen** [comment]
@rrrene Wow, big performance boost comparing to 1.4! ~3 times faster on our project!
